### DISCUSSION TITLE
Descarga masiva de facturas emitidas

### DISCUSSION THREAD

**Discussion by cesarlol** [SAT-CFDI/python-satcfdi#16] [Q&A]
Hola, cómo debo usar el recover_comprobante_iwait para descargar facturas emitidas?

**Comment by Atjoms** [comment]
hay un ejemplo en la documentacion  https://satcfdi.readthedocs.io/en/stable/pages/getting_started/45_cfdi_descarga_massiva.html
# En una solo operación se descargan todos los CFDI de un RFC receptor
# No se recomienda cuando el servicio tome mucho tiempo en responder
for paquete_id, data in sat_service.recover_comprobante_iwait(
        fecha_inicial=date(2020, 1, 1),
        fecha_final=date(2020, 12, 1),
        rfc_receptor=sat_service.signer.rfc,
        tipo_solicitud=TipoDescargaMasivaTerceros.CFDI
):
    with open(f"{paquete_id}.zip", "wb") as f:
        f.write(data)
        
No se recomienda si la pagina esta muy saturada ya que el proceso de la solicitud puede tardar varios dias en algunas ocasiones

**Reply by jicastaneda** [reply]
Hola. Me basé en el ejemplo de tu documentación, y para descarga masiva de facturas de proveedores funciona maravillosamente. Pero creo que lo que nos gustaría saber es cómo podemos descargar masivamente las **facturas emitidas**. Es decir, las facturas de nuestros clientes.

Es posible eso? Qué cambio se tendría que hacer aquí?

`# Facturas Recibidas
response = sat_service.recover_comprobante_request(
    fecha_inicial=date(2020, 1, 1),
    fecha_final=date(2020, 12, 1),
    rfc_receptor=sat_service.signer.rfc,
    tipo_solicitud=TipoDescargaMasivaTerceros.CFDI
)`

Muchísimas gracias de antemano, y los felicito mucho por el desarrollo que hicieron. Me gustaría agradecerles económicamente hablando!

**Reply by josbeporcayo** [reply]
Lo que yo he hecho es primero descargar la metadata con recover_comprobante_request y después mandar solicitudes para cada una. Puedes mandar solicitudes y después checar cuáles están ya listas para descargarse. He recuperado hasta 5 años sin tardarme mucho tiempo. Además para descargar facturas emitidas, solo cambia la opción dentro de la función a rfc_emisor=sat_service.signer.rfc

**Reply by jicastaneda** [reply]
Pensé que sería demasiado bueno para ser verdad ese simple cambio! Excelente @josbeporcayo, lo intentaré. Muchas gracias!!

**Comment by jagm1421** [comment]
Hola, buena tarde, Han podido descargar con las modificaciónes del sat? Ya no me permite desde hace dos semanas
